scalanet/test/src/io/iohk/scalanet/peergroup/kademlia/KRouterSpec.scala

```diff
@@ -206,10 +219,207 @@ class KRouterSpec extends FreeSpec {
         })
       }
     }
+
+    "should do proper initial lookup" - {
+      "when starting with one bootstrap node without neighbours" in {
+        val initialKnownNode = NodeData.getBootStrapNode(0)
+        val result =
+          createTestRouter(peerConfig = Map.empty + (initialKnownNode.myData.id -> initialKnownNode)).runSyncUnsafe()
+
+        result.nodeRecords.size shouldEqual 2
+        result.nodeRecords.get(initialKnownNode.id) shouldBe Some(initialKnownNode.myData)
+      }
+
+      "when starting with 4 bootstrap nodes without neighbours" in {
+        val initialKnownNode = NodeData.getBootStrapNode(0)
+        val initialKnownNode1 = NodeData.getBootStrapNode(0)
+        val initialKnownNode2 = NodeData.getBootStrapNode(0)
+        val initialKnownNode3 = NodeData.getBootStrapNode(0)
+        val initialNodes = Seq(initialKnownNode, initialKnownNode1, initialKnownNode2, initialKnownNode3)
+
+        val result =
+          createTestRouter(
+            peerConfig = Map.empty ++ Seq(
+              initialKnownNode.myData.id -> initialKnownNode,
+              initialKnownNode1.myData.id -> initialKnownNode1,
+              initialKnownNode2.myData.id -> initialKnownNode2,
+              initialKnownNode3.myData.id -> initialKnownNode3
+            )
+          ).runSyncUnsafe()
+
+        result.nodeRecords.size shouldEqual 5
```

Comment:
Think it would be a bit safer if these testcases were in terms of `KRouter.get` (rather than `KRouter.nodeRecords.get`).